### ISSUE TITLE
Problem using the API (bcel-findbugs)

### ISSUE THREAD

**Issue by BoltyDawg** [InventivetalentDev/NickNamer#72]
## What steps will reproduce the problem?  
1. Adding NickNamer as a dependency of plugin using maven
## What were you expecting to happen? What happened instead?  
I used the Maven code provided on the page to build it, but when I do so I get an error stating: ".../repository/com/google/code/findbugs/bcel-findbugs/6.0/bcel-findbugs-6.0.jar' cannot be read or is not a valid ZIP file"
## What version of the plugin are you using?
3.14.1 - I tried other version too
## What Spigot version are you using?
1.12.2
## Additional information?  *(Are you using Bungeecord? Did it work in previous versions? etc.)*
I added the dependency for bcel-findbugs but it still doesn't work

**Comment by BoltyDawg** [comment]
I just copy pasted the dependency straight off the plugin page, but I'll put what I have here: 
```
<dependency>
		<groupId>org.inventivetalent.nicknamer</groupId>
		<artifactId>api</artifactId>
		<version>3.14.1-SNAPSHOT</version>
	</dependency>
```

**Comment by BoltyDawg** [comment]
Okay so I just went through my dependency hierarchy, and after excluding bcel-findbugs and mongodb-driver it compiles okay. But naturally, actually using the API results in an UnsupportedOperationException so that's no good.
I'm not sure if this is related, but whenever I start up the server I get this message:  
`[SpigetUpdate] Invalid SemVer versions specified [3141-SNAPSHOT] [3140-SNAPSHOT]`

**Comment by InventivetalentDev** [comment]
Most of the API methods will throw an UnsupportedOperationException, because the api itself isn't able to handle those methods. You need to install the NickNamer plugin on your server to actually add any functionality to those methods.
So just use the api dependency in your project and then install the plugin and you should be fine.

**Comment by BoltyDawg** [comment]
NickNamer is installed and functional on my server though.
My plugin wouldn't even build until I added the exclusions on the mongodb-driver and bcel-findbugs archives, which apparently causes the API to lose functionality. Do I need to add separate dependencies for these two? I tried doing that and it still didn't work. This is verbatim the error I get when trying to build my plugin normally (my plugin is RPMagic):
`Archive for required library: 'C:/Users/Jay/.m2/repository/com/google/code/findbugs/bcel-findbugs/6.0/bcel-findbugs-6.0.jar' in project 'RPMagic' cannot be read or is not a valid ZIP file	RPMagic - Build path - Build Path Problem`
